apps/prairielearn/src/ee/lib/billing/plans-types.ts

```diff
@@ -0,0 +1,46 @@
+export const PLAN_FEATURE_NAMES = [
+  'course-instance-access',
+  'external-grading',
+  'workspaces',
+] as const;
+export const PLAN_NAMES = ['basic', 'compute', 'everything'] as const;
+
+export type PlanFeatureName = (typeof PLAN_FEATURE_NAMES)[number];
+export type PlanName = (typeof PLAN_NAMES)[number];
+
+interface Plan {
+  features: PlanFeatureName[];
+  initialEnrollmentLimit?: number;
+}
+
+export const PLANS = {
+  // Enabled when student-pays is enabled for a course instance.
+  basic: {
+    features: ['course-instance-access'],
+  },
+  // Add-on to basic plan that enables workspaces and external grading.
```

Comment:
This can actually be used without the `basic` plan (for instance: institution is paying only for basic, course wants to enable compute for their students).